src/abstractdataframe/abstractdataframe.jl

```diff
@@ -1280,3 +1283,168 @@ julia> ncol(df)
 
 """
 (nrow, ncol)
+
+"""
+    disallowmissing(df::AbstractDataFrame,
+        cols::Union{ColumnIndex, AbstractVector, Regex, Not, Colon}=:)
+
+Create a new `DataFrame` in which columns `cols` of data frame `df` are converted
+from element type `Union{T, Missing}` to `T` to drop support for missing values.
+
+If `cols` is omitted all columns in the data frame are converted.
+
+**Examples**
+
+```jldoctest
+julia> df = DataFrame(a=Union{Int,Missing}[1,2])
+2×1 DataFrame
+│ Row │ a      │
+│     │ Int64⍰ │
+├─────┼────────┤
+│ 1   │ 1      │
+│ 2   │ 2      │
+
+julia> disallowmissing(df)
+2×1 DataFrame
+│ Row │ a     │
+│     │ Int64 │
+├─────┼───────┤
+│ 1   │ 1     │
+│ 2   │ 2     │
+```
+"""
+function Missings.disallowmissing(df::AbstractDataFrame,
+                         cols::Union{ColumnIndex, AbstractVector, Regex, Not, Colon}=:)
+    idxcols = Set(index(df)[cols])
+    newcols = AbstractVector[]
+    for i in axes(df, 2)
+        x = df[!, i]
+        if i in idxcols
+            y = disallowmissing(x)
+            push!(newcols, y === x ? copy(y) : y)
+        else
+            push!(newcols, copy(x))
+        end
+    end
+    DataFrame(newcols, _names(df), copycols=false)
+end
+
+"""
+    allowmissing(df::AbstractDataFrame,
+        cols::Union{ColumnIndex, AbstractVector, Regex, Not, Colon}=:)
+
+Create a new `DataFrame` in which columns `cols` of data frame `df` are converted
+to element type `Union{T, Missing}` from `T` to allow support for missing values.
+
+If `cols` is omitted all columns in the data frame are converted.
+
+**Examples**
+
+```jldoctest
+julia> df = DataFrame(a=[1,2])
+2×1 DataFrame
+│ Row │ a     │
+│     │ Int64 │
+├─────┼───────┤
+│ 1   │ 1     │
+│ 2   │ 2     │
+
+julia> allowmissing(df)
+2×1 DataFrame
+│ Row │ a      │
+│     │ Int64⍰ │
+├─────┼────────┤
+│ 1   │ 1      │
+│ 2   │ 2      │
+```
+"""
+function Missings.allowmissing(df::AbstractDataFrame,
+                         cols::Union{ColumnIndex, AbstractVector, Regex, Not, Colon}=:)
+    idxcols = Set(index(df)[cols])
+    newcols = AbstractVector[]
+    for i in axes(df, 2)
+        x = df[!, i]
+        if i in idxcols
+            y = allowmissing(x)
+            push!(newcols, y === x ? copy(y) : y)
+        else
+            push!(newcols, copy(x))
+        end
+    end
+    DataFrame(newcols, _names(df), copycols=false)
+end
+
+"""
+    categorical(df::AbstractDataFrame,
+                cols::Union{ColumnIndex, AbstractVector, Regex, Not, Colon}=:;
```

Comment:
This isn't completely correct, since passing `:` explicitly gives a different result. Maybe (in another PR) we should change that, and require people to pass `categorical(df, String)` or `categorical(df, Union{Missing,String})` explicitly to get the current behavior?

Reply:
I have fixed the docs to make it clearer (and I intentionally put the difference between `:` and no argument in the Examples section earlier).

Now the question is if we find the behavior of `categorical(df)` useful. I find it OK (i.e. when doing e.g. GLM later this is what you want to do) so I think we can leave it as is. But I guess it was you who created the initial implementation - right - so I do not want to override your intentions here 😄.

If we found `categorical(df)` unwanted (as you suggest) then I would add a deprecation warning in this PR and add a keyword argument that would allow the user to specify which types should be converted (but I think it is too complex - almost no one would use it, and the default behavior that we currently have seems to fit 95% of cases and if someone wants something else one simply will pass a list of columns t convert).

Reply:
Fine with me. But better not show `cols=:`, and list two signatures instead, otherwise it's really misleading (the real value would be `[eltype(col) <: Union{AbstractString, Missing} for col in eachcol(df)]`).

Reply:
This is exactly what I did (see the PR after I pushed).